src/states/elder/mod.rs

```diff
@@ -919,7 +925,7 @@ impl Elder {
     }
 
     fn maintain_parsec(&mut self) {
-        if self.parsec_map.needs_pruning() {
+        if self.parsec_map.needs_pruning() && !self.chain.is_churn_in_progress() {
```

Comment:
This is probably not where we should check for `!self.chain.is_churn_in_progress()` but probably in `handle_prune`.
Additionally, we need to make sure that ParsecPrune event are not carried over when a new parsec is created.

Reply:
`handle_prune` is kind of a local state, which means once received consensused ParsecPrune, different node may carry out or not to carry out the pruning, depends on its churn_in_progress status.
Meanwhile, the `voting for ParsecPrune` is a shared state, and once a node received the consensused vote. The action shall be carried out.

> ParsecPrune event are not carried over when a new parsec is created
`ParsecPrune` will be filterred out when carry out revote, within reset_parsec_with_data

Reply:
Sorry I do not understand `handle_prune is kind of a local...` and `Meanwhile, the voting for ParsecPrune is a sh...` sentences.

Reply:
The current approach may make some nodes vote for ParsecPrune, some not. However, once the ParsecPrune got consensused. They all will carry out the pruning.

If we move the check of `self.chain.is_churn_in_progress()` into `handle_churn`, some nodes may decide to carry out the pruning, some nodes may decide not.

Reply:
I do not think so, as is_rn_in_progress() should be shared state so all should do the same.
But mostly, if you all voted for prune, and now churn is happening, and now Prune get consensus, we should not prune. so it would seem incorrect to not check in handle_prune whether we should carry out the prune or not (we do check for is split in there already).

Reply:
right.
then shall we do a double check? that is:
keep the check here, and in `handle_prune` carries out one more check?

Reply:
Probably, not, what would be the purpose of this one with there is one at the end?

Reply:
avoiding the extra vote (which incurs traffic) when there is churn already happens?

Reply:
A single vote does not seem to matter much, there will be other traffic going on.

Reply:
ok then. I will move it to only check inside handle_churn